File: 프로그래머스/연습문제/Lv1/가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> idx(26, -1);
    vector<int> answer;
    for(int i=0; i<s.length(); i++){
        if(idx[s[i]-'a'] == -1) answer.push_back(-1);
        else answer.push_back(i-idx[s[i]-'a']);
        
        idx[s[i]-'a'] = i;
    }
    
    return answer;
}

File: 프로그래머스/연습문제/Lv1/가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    idx = [-1 for i in range(26)]
    answer = []
    for i in range(len(s)):
        if idx[ord(s[i])-ord('a')] == -1:
            answer.append(-1)
        else:
            answer.append(i-idx[ord(s[i])-ord('a')])

        idx[ord(s[i])-ord('a')] = i

    return answer
